all.go

```diff
@@ -129,72 +129,43 @@ func (n *Node) All(to interface{}, options ...func(*index.Options)) error {
 }
 
 func (n *Node) all(tx *bolt.Tx, info *modelInfo, ref *reflect.Value, rtyp, typ reflect.Type, opts *index.Options) error {
+	var err error
 	results := reflect.MakeSlice(reflect.Indirect(*ref).Type(), 0, 0)
 	bucket := n.GetBucket(tx, info.Name)
 
 	if bucket != nil {
 		c := bucket.Cursor()
 		if opts.Reverse {
+			// loop through the records in descending order
 			for k, v := c.Last(); k != nil; k, v = c.Prev() {
-				if v == nil {
+				c, b := loopControl(v, opts)
```

Comment:
`c` shadows the cursor, and it is not clear at all what is `c` and what is `b` without looking at the implementation of `loopControl`. Also `loopControl` is a name too generic: what does `loopControl` ? can i call it outside of a loop with the right parameters? The name should be more explicit